Stablegraph edge_indices borrows graph

<!--
  -- Thanks for filing a `petgraph` issue! Fill out the appropriate template
  -- below. We appreciate it :)
  -->

<!-- ************************ BUG REPORTS *************************** -->

### Test Case

```
    let mut test = StableUnGraph::<(),()>::default();
    for i in test.edge_indices() {
        test.remove_edge(i);
    }
```


Compiling this code produces error
 `error[E0502]: cannot borrow test as mutable because it is also borrowed as immutable`
but when test graph is UnGraph
```
    let mut test = UnGraph::<(),()>::default();
    for i in test.edge_indices() {
        test.remove_edge(i);
    }
```
 it compiles no problem, docs page also states that indices allow for mutability during traversing so i believe your intent is for that to work, im really sorry if I'm wrong and this is a desired reaction.
I checked code in StableGraph and found that EdgeIterator borrows whole field Edges and handles finding indexes in .next, it explains everything so in the end i did

```
    let mut test = StableUnGraph::<(), ()>::default();
    for i in test.edge_indices().collect::<Vec<_>>() {
        test.remove_edge(i);
    }
```
it works, but i believe this should be changed or adressed in docs

--- a/src/graph_impl/stable_graph/mod.rs
+++ b/src/graph_impl/stable_graph/mod.rs
@@ -629,7 +629,10 @@ where
         }
     }
 
-    /// Return an iterator over the edge indices of the graph
+    /// Return an iterator over the edge indices of the graph.
+    ///
+    /// Note: the iterator borrows a graph in contrast to the behavior of
+    /// [`Graph::edge_indices`](fn@crate::Graph::edge_indices).
     pub fn edge_indices(&self) -> EdgeIndices<E, Ix> {
         EdgeIndices {
             iter: enumerate(self.raw_edges()),
@@ -1803,6 +1806,8 @@ impl<N, Ix: IndexType> DoubleEndedIterator for NodeIndices<'_, N, Ix> {
 }
 
 /// Iterator over the edge indices of a graph.
+///
+/// Note: `EdgeIndices` borrows a graph.
 #[derive(Debug, Clone)]
 pub struct EdgeIndices<'a, E: 'a, Ix: 'a = DefaultIx> {
     iter: iter::Enumerate<slice::Iter<'a, Edge<Option<E>, Ix>>>,

